lib/experimental/media/load.php

```diff
@@ -85,26 +85,21 @@ function gutenberg_media_processing_filter_rest_index( WP_REST_Response $respons
 
 add_filter( 'rest_index', 'gutenberg_media_processing_filter_rest_index' );
 
-
 /**
  * Overrides the REST controller for the attachment post type.
  *
  * @param array  $args      Array of arguments for registering a post type.
  *                          See the register_post_type() function for accepted arguments.
- * @param string $post_type Post type key.
  */
-function gutenberg_filter_attachment_post_type_args( array $args, string $post_type ): array {
-	if ( 'attachment' === $post_type ) {
-		require_once __DIR__ . '/class-gutenberg-rest-attachments-controller.php';
+function gutenberg_filter_attachment_post_type_args( array $args ): array {
+	require_once __DIR__ . '/class-gutenberg-media-processing-rest-attachments-controller.php';
 
-		$args['rest_controller_class'] = Gutenberg_REST_Attachments_Controller::class;
-	}
+	$args['rest_controller_class'] = Gutenberg_Media_Processing_REST_Attachments_Controller::class;
 
 	return $args;
 }
 
-add_filter( 'register_post_type_args', 'gutenberg_filter_attachment_post_type_args', 10, 2 );
```

Comment:
The register_attachment_post_type_args filter is more specific and runs after the general register_post_type_args filter.

This ensures the integrity of the inheritance chain:

```
WP_REST_Attachments_Controller
    ↓
Gutenberg_REST_Attachments_Controller (WordPress 6.9 compatibility)
    ↓
Gutenberg_Media_Processing_REST_Attachments_Controller (Media experiment)
```

cc @swissspidy or @adamsilverstein for a confidence check and whether you needed `register_post_type_args` specifically. I smoke tested the media processing experiment and couldn't find any issues.